server.go

```diff
@@ -822,34 +822,26 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 	// Based on the disk representation of the node announcement generated
 	// above, we'll generate a node announcement that can go out on the
 	// network so we can properly sign it.
-	nodeAnn, err := selfNode.NodeAnnouncement(false)
+	s.currentNodeAnn, err = selfNode.NodeAnnouncement(false)
 	if err != nil {
 		return nil, fmt.Errorf("unable to gen self node ann: %v", err)
 	}
 
-	// With the announcement generated, we'll sign it to properly
-	// authenticate the message on the network.
-	authSig, err := netann.SignAnnouncement(
-		s.nodeSigner, nodeKeyDesc.KeyLocator, nodeAnn,
-	)
+	// With the announcement generated, we'll use our getNodeAnnouncement
+	// helper to sign our current announcement to properly authenticate
+	// with the network.
+	nodeAnn, err := s.genNodeAnnouncement(true)
```

Comment:
Just observing that this now updates the timestamp, and I think that didn't happen previously. Not sure if it matters?